src/schedule/ucc_schedule.c

```diff
@@ -122,11 +123,15 @@ ucc_schedule_completed_handler(ucc_coll_task_t *parent_task, //NOLINT
                                ucc_coll_task_t *task)
 {
     ucc_schedule_t *self = ucc_container_of(task, ucc_schedule_t, super);
+    uint32_t        n_completed_tasks;
 
-    // TODO: do we need lock here?
-    // if tasks in schedule are independet and completes concurently
-    self->n_completed_tasks += 1;
-    if (self->n_completed_tasks == self->n_tasks) {
+    if (UCC_TASK_THREAD_MODE(task) == UCC_THREAD_MULTIPLE) {
```

Comment:
probably you are right, especially since this is for algorithms with "schedules", ie smth which already has more instructions and more heavy. i'll run just a quick smoke perf test and update

Reply:
measured. i see no diff, so moved to always use atomic as suggested